src/socket.c

```diff
@@ -648,12 +687,15 @@ static int pthreads_sockets_from_fd_set(zval *sockets, fd_set *fds) /* {{{ */
 }
 /* }}} */
 
-void pthreads_socket_select(zval *read, zval *write, zval *except, uint32_t sec, uint32_t usec, zval *return_value) {
+void pthreads_socket_select(zval *read, zval *write, zval *except, uint32_t sec, uint32_t usec, zval *errorno, zval *return_value) {
 	fd_set rfds, wfds, efds;
 	php_socket_t mfd = 0;
 	int result = SUCCESS, sets = 0;
 	struct timeval  tv;
 
+	zval_dtor(errorno);
```

Comment:
This should be [`zval_ptr_dtor`](https://phpinternals.net/docs/zval_ptr_dtor), otherwise it will leak on cyclic references.

Reply:
Additionally this will segfault if the `errorno` parameter is not given.

Reply:
I've copied this part from https://github.com/krakjoe/pthreads/blob/d32079fb4a88e6e008104d36dbbf0c2dd7deb403/src/socket.c#L775. Is `zval_dtor` in `pthreads_socket_recvfrom` also faulty?

Reply:
Ok. `zval_dtor` was replaced by `zval_ptr_dtor`. Maybe you could have a look at the original implementation of `socket_recvfrom` https://github.com/php/php-src/blob/master/ext/sockets/sockets.c#L1771 which uses `zval_dtor` too.

Reply:
Ok, I'll have a look into this, thanks.